src/mapping_navigation/scripts/pure_pursuit.py

```diff
@@ -0,0 +1,143 @@
+"""
+Code adapted from
+
+Path tracking simulation with pure pursuit steering and PID speed control.
+author: Atsushi Sakai (@Atsushi_twi)
+        Guillaume Jacquenot (@Gjacquenot)
+"""
+import numpy as np
+import math
+from typing import List, Tuple
+from map_information import *
+
+class State:
+
+    """
+    Contains the car information needed in pure pursuit
+    The cartesian coordinates for location
+    The yaw of the car
+    The cars wheel base
+    """
+    def __init__(self, wheel_base: float, point: Point=Point((0,0)), yaw:float=0.0, velocity:float=0.0):
+        self.wheel_base = wheel_base
+        self.point = point
+        self.yaw = yaw
+        self.velocity = velocity
+        self.rear_x = self.point.coordinate[0] - ((wheel_base / 2) * math.cos(self.yaw))
+        self.rear_y = self.point.coordinate[1] - ((wheel_base / 2) * math.sin(self.yaw))
+    
+    """
+    Updates the cars position
+    Cartesian coordinates for the cars position
+    Quaterion to find the yaw of the car
+    """
+    def update_pos(self, point: Point, quaterion: Tuple[float, float, float, float]):
+        self.point = point
+        self.yaw = euler_from_quaternion(quaterion)
+        self.rear_x = self.point.coordinate[0] - ((self.wheel_base / 2) * math.cos(self.yaw))
+        self.rear_y = self.point.coordinate[1] - ((self.wheel_base / 2) * math.sin(self.yaw))
+
+    """
+    Updates the cars speed
+    """
+    def update_speed(self, velocity:float):
+        self.velocity = velocity
+
+    """
+    Find the distance between the cars current location and another point
+    Find the euclidian distance
+    """
+    def calc_distance(self, point_x:float, point_y:float) -> float:
+        dx = self.rear_x - point_x
+        dy = self.rear_y - point_y
+        return math.hypot(dx, dy)
+
+class PurePursuit:
```

Comment:
Is this class unit-testable? If so we should unit test it. It's probably not easy to detect if something like this breaks.

Reply:
Yeah that's mostly why I left PurePursuit as its own class and not as a part of navigation. For the moment I would like to get this PR in without the units test mainly cause I am not sure if we want to use a specific python testing framework, but I agree we should probably move towards including units test with each feature were possible. Three easy unit test to use for this one is see if it selects the correct goal point at low and high speed and also make sure its getting around the correct angle.